### PR TITLE
Made it so we clean up gl resources when view controllers get deleted.

### PR DESCRIPTION
**Pull request by gaaclarke**
Relevant issue: https://github.com/flutter/flutter/issues/38736

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -203,6 +203,12 @@ - (void)setFlutterViewControllerWillDeallocObserver:(id<NSObject>)observer {
 - (void)notifyViewControllerDeallocated {
   if (!_allowHeadlessExecution) {
     [self destroyContext];
+  } else {
+    PlatformViewIOS platform_view = [self iosPlatformView];
+    if (platform_view) {
+      fml::WeakPtr<FlutterViewController> null_view_controller;
+      platform_view->SetOwnerViewController(null_view_controller);
```
Comment (dnfield) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: Can this just be `platform_view->SetOwnerViewController({});`?

**Comment by xster** [comment]
Oh, pretty funny that the PR got so much smaller :D

Can we add some tests for this?

**Comment by dnfield** [comment]
There's still value in doing the refactor but for now this should fix the bug.

Testing this is really hard right now, see discussion on previous pr

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm (hunk 1 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: done

**Comment by xster** [comment]
LGTM

**Pull request merged**